CMS-Enterprise/sbom-harbor
# refactor: improve paths and filenames when storing

## Summary

This PR updates the storage providers to address a storage paths and fixing some issues in the 

### Changed

- Files/objects should have a .json extension
- Files/objects should be saved to a folder/prefix

### Fixed

- File names should no longer ever have double hypens
- File names should never begin or end with a hypen
- Made `Sbom::created` field optional which was causing a backward compatability issue
- Made `SbomService::storage` optional which was also causing a compatability issue.

## How to test

- Pull latest
- Start the devenv
- Make sure you have data in the local mongo (run ingest & enrich if you don't)
- `cargo build`
- Run `./target/debug/harbor analyze --debug sbom-detail` 
- Run `ls /tmp/harbor-debug/analyze/sbom-detail/analytic-detailed-report` to make sure that file are present in that path and named correctly.
